### PR TITLE
RCT/YJ/are_all_terminal_types_VAV function

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I couldn't modify the RDS since it hasn't been merged. I'll make a few minor changes once it's merged. Thanks!

### CODE REVIEW

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def are_all_terminal_types_VAV(rmd: dict, terminal_unit_id_list: list[str]) -> bool:
+    """
+    Returns TRUE if all of the terminal unit types input to this function are variable air volume (VAV) or are equal to null.
+    It returns FALSE if any of the terminal units are of a type other than variable air volume (VAV) or null.
+
+    Parameters
+    ----------
+    rmd: json,
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:14]: I would recommend changing the type hint for rmd in the docstring to dict to align with the function description.

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        To evaluate if the terminal unit types are variable air volume (VAV).
+    terminal_unit_id: list[str],
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:16]: Also `terminal_unit_id` to `terminal_unit_id_list`

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+def test__TEST_RMD__is_valid():
+    schema_validation_result = schema_validate_rmd(TEST_RMD)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__hvac_fan_all_VAV__success():
+    assert are_all_terminal_types_VAV(TEST_RMI, ["terminal_1", "terminal_4"]) is True
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py:68]: By looking at the test case, I'm wondering in `are_all_terminal_types_VAV(rmd: dict, terminal_unit_id_list: list[str])`, if `rmd `should be called `rmi`

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def test__hvac_fan_not_all_VAV__success():
+    assert (
+        are_all_terminal_types_VAV(TEST_RMI, ["terminal_1", "terminal_2", "terminal_4"])
+        is False
+    )
+
+
+def test__hvac_fan_include_no_type__success():
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py:78]: I'm curious about what if there is no terminal in the input rmd.

**Review by Jiarongx-Xie — COMMENTED** [review]
Thank you Yun Joon for finishing this dev. I left several minor comments.

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:14]: Addressed

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV_test.py:78]: In this scenario, the results will be false.

**Comment by weilixu** [comment]
Don't we already have this function in `ruleset_functions/baseline_systems/baseline_hvac_sub_functions`? I thought it is a modification not a new function.

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    # check `terminal_unit_id_list` type
+    assert_(
+        isinstance(terminal_unit_id_list, list)
+        and all(
+            isinstance(terminal_unit_id, str)
+            for terminal_unit_id in terminal_unit_id_list
+        ),
+        "Please make sure the `terminal_unit_id_list` type is list and has list of strings",
+    )
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:33]: I think the type checking will ensure the type. We do not need to add the assert here.

The return statement also handles an empty list scenario. So it should be fine.

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    return all(
+        [
+            find_exactly_one_terminal_unit(rmd, terminal_id).get("type") is None
+            or find_exactly_one_terminal_unit(rmd, terminal_id).get("type")
+            == TERMINAL_TYPE.VARIABLE_AIR_VOLUME
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:39]: can do `find_exactly_one_terminal_unit(rmd, terminal_id).get("type") in [None, TERMINAL_TYPE.VARIABLE_AIR_VOLUME]`

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    assert_(
+        isinstance(terminal_unit_id_list, list)
+        and all(
+            isinstance(terminal_unit_id, str)
+            for terminal_unit_id in terminal_unit_id_list
+        ),
+        "Please make sure the `terminal_unit_id_list` type is list and has list of strings",
+    )
+
+    return all(
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:35]: should add: `len(terminal_unit_id_list) > 0 and all(...)`

File: rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+from rct229.utils.assertions import assert_
+from rct229.utils.utility_functions import find_exactly_one_terminal_unit
+
+TERMINAL_TYPE = SchemaEnums.schema_enums["TerminalOptions"]
+
+
+def are_all_terminal_types_VAV(rmd: dict, terminal_unit_id_list: list[str]) -> bool:
+    """
+    Returns TRUE if all of the terminal unit types input to this function are variable air volume (VAV) or are equal to null.
+    It returns FALSE if any of the terminal units are of a type other than variable air volume (VAV) or null.
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/are_all_terminal_types_VAV.py:11]: ` It returns FALSE if no terminal unit in the list or any of the terminal units are of a type other than variable air volume (VAV) or null.`

**Comment by yunjoonjung-PNNL** [comment]
This PR was replaced with PR #1419. Also, addressed Weili's comments in the PR.

**Pull request closed without merging**